packages/shared/providers.yaml

```diff
@@ -1318,6 +1318,11 @@ linkhut:
     proxy:
         base_url: https://api.ln.ht
     docs: https://docs.nango.dev/integrations/all/linkhut
+mailgun:
+    categories:
+        - marketing
+    auth_mode: API_KEY
```

Comment:
I feel like it's a bit more complicated than just this. API_KEY means `Authorization Bearer ${apiKey}` but they are actually using a fake Basic auth with an api key, meaning `Authorization ${base64(user:apikey)}`

https://documentation.mailgun.com/docs/mailgun/api-reference/intro/

https://github.com/mailgun/mailgun.js/blob/308717a7f70b31617afd3a427b0ee3859b8aafbe/lib/Classes/common/Request.ts#L121

Not sure if we handle that? and if not how? @khaliqgant @hassan254-prog

Reply:
imo, the Basic auth_mode should work here. According to the [docs](https://documentation.mailgun.com/docs/mailgun/api-reference/intro/#authentication), the curl command for authentication is `curl --user 'api:YOUR_API_KEY'` which will add `Authorization: Basic ${base64(api:YOUR_API_KEY)}` to the header where `api` is the username and `YOUR_API_KEY` the password when creating a new connection in Nango. I guess the best way to know if it will work is by actually testing the implementation. 
@jwd-dev  could you test the following and provide us with any feedback you get. 

```
mailgun:
    categories:
        - marketing
    auth_mode: BASIC
     proxy:
        base_url: https://${connectionConfig.region}.mailgun.net
        verification:
            method: GET
            endpoint: /v4/domains
    docs: https://docs.nango.dev/integrations/all/mailgun
```


When adding a new connection, `api` is the region for US-based customers, and `api.eu` is the region for EU-based customers.

Reply:
I handle the fake basic auth on my side.

Reply:
Tested this and it works, just pushed my changes.